pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -632,7 +637,9 @@ private void createPersistentTopic(final String topic, boolean createIfMissing,
         }
 
         getManagedLedgerConfig(topicName).thenAccept(managedLedgerConfig -> {
-            managedLedgerConfig.setCreateIfMissing(createIfMissing);
+            managedLedgerConfig.setCreateIfMissing(useDefaultConfig ? 
```

Comment:
This passing two booleans down a whole tree of methods only to compare them is spagetti. Instead, at the callsite where createIfMissing would be true, check default config there (wrap in a method if there's a few of them).